### PR TITLE
NMS-13726: Fix JMS twin issues

### PR DESCRIPTION
**Pull request by cgorantla**
Use topic for Twin Sink
Add authorization entries for Twin topics

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13726

### CODE REVIEW

**Comment by cgorantla** [comment]
Still validating activemq changes for Twin stuff.
Tested queue -> topic changes.

File: core/ipc/twin/common/src/main/java/org/opennms/core/ipc/twin/common/AbstractTwinSubscriber.java [hunk 1]
```diff
@@ -97,10 +97,9 @@ public AbstractTwinSubscriber() {
             } catch (Exception e) {
                 LOG.error("Exception while sending response to consumer", e);
             }
-        } else {
-            TwinRequest twinRequest = new TwinRequest(key, location);
-            sendRpcRequest(twinRequest);
         }
+        TwinRequest twinRequest = new TwinRequest(key, location);
+        sendRpcRequest(twinRequest);
```
Comment (fooker) [inline core/ipc/twin/common/src/main/java/org/opennms/core/ipc/twin/common/AbstractTwinSubscriber.java:102]: Why is this required?

Comment (cgorantla) [reply core/ipc/twin/common/src/main/java/org/opennms/core/ipc/twin/common/AbstractTwinSubscriber.java]: not needed, removed.

**Comment by j-white** [comment]
Looks reasonable to me.

File: opennms-base-assembly/src/main/filtered/etc/opennms-activemq.xml [hunk 2]
```diff
@@ -50,6 +50,10 @@
                   <authorizationEntry queue="OpenNMS.*.*" write="minion" admin="minion" />
                   <!-- Users in the minion role can read/create from queues that are keyed by location -->
                   <authorizationEntry queue="OpenNMS.*.*.*" read="minion" admin="minion" />
+                    <!-- Users in the minion role can read queue OpenNMS.Twin.Sink -->
+                    <authorizationEntry queue="OpenNMS.Twin.Sink" read="minion" admin="minion" />
+                    <!-- Users in the minion role can write queue OpenNMS.Twin.RPC -->
+                    <authorizationEntry queue="OpenNMS.Twin.RPC" write="minion" admin="minion" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/opennms-activemq.xml:56]: Do we have here a formatting issue?

**Review by indigo423 — APPROVED** [review]
Tested in my environment and the error messages in the eventd.log like these are gone now.

```
2021-11-10 10:12:46,034 WARN  [ActiveMQ Transport: tcp:///172.23.42.2:52590@61616] o.a.a.b.T.Service: Security Error occurred on connection to: tcp://172.23.42.2:52590, User minion is not authorized to read from: queue://OpenNMS.Twin.Sink
```

**Comment by indigo423** [comment]
@cgorantla What was wondering, why did this ActiveMQ transport error appear in the eventd.log?

**Comment by cgorantla** [comment]
> @cgorantla What was wondering, why did this ActiveMQ transport error appear in the eventd.log?
created https://issues.opennms.org/browse/NMS-13731 to handle this.
